### DISCUSSION TITLE
Type: AirPurifier possible?

### DISCUSSION THREAD

**Discussion by mhunt407** [milo526/homebridge-tuya-web#192] [Q&A] [answered]
Hi there! I wondered if someone more techy than I could help - I've got an air purifier that works through Tuya/Smart Life. 

I discovered that functionality to add them and adjust fan speed/show air quality was possible through another plugin with the following options, but you need ID and Key which you can't get at the moment.

I wondered if it was possible using this plugin? I can't take any credit for this, but the config looked like:

"name": "Living Room Air Purifier",
 "type": "AirPurifier",
"manufacturer": "Elechomes",
"model": "KJ200G-A3B-UK",
"id": "REDACTED",
"key": "REDACTED",
"ip": "192.168.99.50",
"version": "3.3",
"fanSpeedSteps": 3,
"showAirQuality": true

Thanks!

**Comment by milo526** [comment, marked as answer]
Sadly this is not possible.

The plugin that you mention uses a different API. This API supports more devices (such as Air Purifiers) but also requires more setup and configuration (such as the ID and key).

Since I do not work for Tuya I cannot change the implementation of the API that this plugin uses.
This means that it will not be possible to control your device without the id and key.
